src/components/modules/api/toolbar.ts

```diff
@@ -31,4 +32,36 @@ export default class ToolbarAPI extends Module {
   public close(): void {
     this.Editor.Toolbar.close();
   }
+
+  /**
+   * Toggles Block Setting of the current block
+   *
+   * @param {boolean} openingState —  opening state of Block Setting
+   */
+  public toggleBlockSettings(openingState?: boolean): void {
+    if (!this.Editor.BlockManager.currentBlockIndex) {
+      _.logLabeled('Could\'t toggle the Toolbar because there is no block selected ', 'warn');
+
+      return;
+    }
+
+    if (!this.Editor.Toolbar.opened) {
+      this.Editor.Toolbar.open(true, false);
+      this.Editor.Toolbar.plusButton.hide();
+    }
+
+    /** Check that opening state is set or not */
+    const canOpenBlockSettings = (openingState !== undefined) ? openingState : !this.Editor.BlockSettings.opened;
+
+    /** Check if state same as current state */
+    if (openingState !== undefined && openingState === this.Editor.BlockSettings.opened) {
```

Comment:
`openingState !== undefined` is unused. 

```suggestion
    if (canOpenBlockSettings/* or openingState */ === this.Editor.BlockSettings.opened) {
```

Reply:
Done